### DISCUSSION TITLE
Allow Astro.slot.render() to also support raw passed in format

### DISCUSSION THREAD

**Discussion by jinsupark** [withastro/roadmap#827] [Proposal]
### Body

# Summary

The Astro.slot.render() function does a great job of rendering slot to html, but it would also be extremely helpful if it could also support rendering the passed in data as-is.

# Background & Motivation

Astro is an extremely useful framework for us because it allows us to render any component regardless of language. We internally use this to build documentation for UI components similar to storybook.

We build a custom docs templates that allow us to pass in the component and render the preview and html code snippet like so:

```
<CodePreviewer>
  <BasicTextField validation="error">
    Hello world
  <BasicTextField />
</CodePreviewer>
  ```
  
  Currently we are successfully able to render both the preview and html code snippet, however we also need the ability to render the unrendered snippet like so:
  
  ```
  <BasicTextField validation="error">
    Hello world
  <BasicTextField />
  ```

# Goals

What we need is the ability to do something like, `Astro.slot.render('default', 'raw')` to pull the raw unrendered slot value.

# Example

It would be great if the Astro.slot.render() could accept a second parameter that defines the format:

`Astro.slot.render('default', 'raw')` or `Astro.slot.render('default', 'html')` (default)

**Comment by david-hodgetts** [comment]
Wanted to quickly chime in to say that at my current workplace, we have the same use case. It would be very handy for us to be able to have access to the raw, _as is_, content of an astro component's slot.

**Comment by louiss0** [comment]
This looks like a good idea.

**Comment by ngdangtu-vn** [comment]
I have a case where I want to create an inline code tag and I prefer to have a raw html content rather then rendering one with additional attributes (such as `data-astro-source-file` and `data-astro-source-loc`). I suggest to have a different method instead of using `render()` because it already use rest params for passing additional argument.

```js
Astro.slots.raw('default')
```

This would be more clear imho.
